python/pyspark/ml/param/shared.py

```diff
@@ -288,6 +288,7 @@ class HasNumFeatures(Params):
 
     def __init__(self):
         super(HasNumFeatures, self).__init__()
+        self._setDefault(numFeatures=1 << 18)
```

Comment:
Do we need a Python default if it's set in Scala?

Reply:
I am not sure, but default values of some other params (like `parallelism`/`aggregationDepth`/`distanceMeasure`) are set in both sides